src/panels/lovelace/components/hui-conditional-base.ts

```diff
@@ -47,27 +54,84 @@ export class HuiConditionalBase extends ReactiveElement {
     this._config = config;
   }
 
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._clearMediaQueries();
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._listenMediaQueries();
+    this._updateVisibility();
+  }
+
+  private _clearMediaQueries() {
+    this._mediaQueries = [];
+    while (this._mediaQueriesListeners.length) {
+      this._mediaQueriesListeners.pop()!();
+    }
+  }
+
+  private _listenMediaQueries() {
+    if (!this._config) {
+      return;
+    }
+
+    const conditions = this._config.conditions.filter(
+      (c) => c.condition === "responsive"
+    ) as ResponsiveCondition[];
+
+    const mediaQueries = conditions
+      .map((c) => c.media_query ?? "")
+      .filter((m) => m !== "");
+
+    if (deepEqual(mediaQueries, this._mediaQueries)) return;
+
+    this._mediaQueries = mediaQueries;
+    while (this._mediaQueriesListeners.length) {
+      this._mediaQueriesListeners.pop()!();
+    }
+    mediaQueries.forEach((query) => {
+      const listener = listenMediaQuery(query, () => {
+        this._updateVisibility();
```

Comment:
Wonder if we can this more efficient, so we don't have to check the media query again in `_updateVisibility` 🤔  can get complicated if there are multiple, but maybe we should optimise for the most common case of 1 media query/condition?